engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -638,46 +655,57 @@ public VolumeInfo createVolume(VolumeInfo volume, VirtualMachine vm, VirtualMach
 
         pool = findStoragePool(dskCh, dc, pod, clusterId, vm.getHostId(), vm, avoidPools);
         if (pool == null) {
-            s_logger.warn("Unable to find suitable primary storage when creating volume " + volume.getName());
-            throw new CloudRuntimeException("Unable to find suitable primary storage when creating volume " + volume.getName());
+            String msg = String.format("Unable to find suitable primary storage when creating volume [%s].", volumeToString);
+            s_logger.error(msg);
+            throw new CloudRuntimeException(msg);
         }
 
+        String poolToString = getReflectOnlySelectedFields(pool);
+
         if (s_logger.isDebugEnabled()) {
-            s_logger.debug("Trying to create " + volume + " on " + pool);
+            s_logger.debug(String.format("Trying to create volume [%s] on storage pool [%s].",
+                    volumeToString, poolToString));
         }
         DataStore store = dataStoreMgr.getDataStore(pool.getId(), DataStoreRole.Primary);
         for (int i = 0; i < 2; i++) {
             // retry one more time in case of template reload is required for Vmware case
             AsyncCallFuture<VolumeApiResult> future = null;
-            boolean isNotCreatedFromTemplate = volume.getTemplateId() == null ? true : false;
+            boolean isNotCreatedFromTemplate = volumeInfo.getTemplateId() == null ? true : false;
```

Comment:
```suggestion
            boolean isNotCreatedFromTemplate = (volumeInfo.getTemplateId() == null);
```

Reply:
not due to your code @paulazomig , but an easy improvement.

Reply:
there are some others, like reducing cyclic complexity and extracting a string constant that would be easy wins. Not a -1, but easy wins they would be.